### PR TITLE
[WIP] migrate log4j from v1.2 to v2.X

### LINKED ISSUE #1592: Howto configure Cygnus Log4J to log files rotated using gzip
Howto configure Cygnus Log4J to log files rotated using gzip

### PR DESCRIPTION
**Pull request by fgalan**
Fixes #1592 

CC: @cesarjorgemartinez 

- [x] Upgrade cygnus to use  API  from log4j v2
- [x] Upgrade log4j configuration files
- [ ]  Test
- [ ]  Test change log level

This PR should be merged with squash

### CODE REVIEW

**Comment by AlvaroVega** [comment]
According with travis build:
```
[ERROR] Failed to execute goal on project cygnus-common: Could not resolve dependencies for project com.telefonica.iot:cygnus-common:jar:1.10.0_SNAPSHOT: Could not find artifact log4j:log4j:jar:2.11.2 in central (http://repo.maven.apache.org/maven2) -> [Help 1]
```

**Comment by IvanHdzC** [comment]
Still not working. 

function PatternLayout.getConversionPattern() deprecated.

Looking for a workaround.

**Comment by fgalan** [comment]
Recently updated with master done by @joelcamus (thanks! :). However, still not working...

**Comment by AlvaroVega** [comment]
i.e.:

time=2022-02-21T14:04:09.131Z | lvl=INFO | corr=156b12c7-a151-4b56-a92c-6860ff6253b1; cbnotif=1 | trans=0ff4c0fa-893d-408c-97aa-75627e3143ff | srv=N/A | subsrv=N/A | comp= | op=processNewBatches | msg=com.telefonica.iot.cygnus.sinks.NGSISink[621] : Finishing internal transaction (156b12c7-a151-4b56-a92c-6860ff6

**Comment by AlvaroVega** [comment]
At this point with this PR logs are:

time=2022-02-23T11:00:32.370Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp= | op=processNewBatches | msg=com.telefonica.iot.cygnus.sinks.NGSISink[528] : Batch accumulation time reached, the batch will be processed as it is
time=2022-02-23T11:00:32.644Z | lvl=DEBUG | corr= | trans= | srv= | subsrv= | comp= | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[205] : [nmi] The configuration has not changed
time=2022-02-23T11:00:32.644Z | lvl=DEBUG | corr= | trans= | srv= | subsrv= | comp= | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[205] : [nmi] The configuration has not changed
time=2022-02-23T11:00:32.653Z | lvl=DEBUG | corr= | trans= | srv= | subsrv= | comp= | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[205] : [nmi] The configuration has not changed
time=2022-02-23T11:00:32.653Z | lvl=DEBUG | corr= | trans= | srv= | subsrv= | comp= | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[205] : [nmi] The configuration has not changed

but logs with previous version were:
time=2022-02-23T11:04:06.850Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[205] : [nmi] The configuration has not changed
time=2022-02-23T11:04:06.882Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[205] : [nmi] The configuration has not changed
time=2022-02-23T11:04:06.887Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[205] : [nmi] The configuration has not changed
time=2022-02-23T11:04:06.914Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[205] : [nmi] The configuration has not changed
time=2022-02-23T11:04:15.685Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=processNewBatches | msg=com.telefonica.iot.cygnus.sinks.NGSISink[527] : Batch accumulation time reached, the batch will be processed as it is

**Comment by AlvaroVega** [comment]
Now it seems all logs are fine:

time=2022-02-24T11:41:00.733Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=getConnection | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl$SQLDriver[924] : MYSQL Pool status (smartcity) Max.: 3; Active: 2; Idle: 0
time=2022-02-24T11:41:00.767Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=insertErrorLog | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl[813] : MYSQL Executing SQL query 'insert into `smartcity_error_log` (timestamp, error, query) values (?, ?, ?)'
time=2022-02-24T11:41:00.778Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=closeSQLObjects | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl[470] : MYSQL Closing SQL connection objects.
time=2022-02-24T11:41:00.778Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=insertErrorLog | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl[823] : MYSQL Trying to add 'smartcity' and 'smartcity_error_log' to the cache after insertion
time=2022-02-24T11:41:00.778Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=addDataBase | msg=com.telefonica.iot.cygnus.backends.sql.SQLCache[56] : 'smartcity' added to the cache
time=2022-02-24T11:41:00.779Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=addTable | msg=com.telefonica.iot.cygnus.backends.sql.SQLCache[76] : 'smartcity_error_log' added to the cache
time=2022-02-24T11:41:00.781Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=getConnection | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl$SQLDriver[903] : MYSQL Recovered destination connection from cache (smartcity)
time=2022-02-24T11:41:00.782Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=getConnection | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl$SQLDriver[924] : MYSQL Pool status (smartcity) Max.: 3; Active: 2; Idle: 0
time=2022-02-24T11:41:00.784Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=purgeErrorTable | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl[774] : MYSQL Executing SQL query 'delete from `smartcity_error_log` where timestamp not in (select timestamp from (select timestamp from `smartcity_error_log` order by timestamp desc limit 100 ) tmppurge )'
time=2022-02-24T11:41:00.786Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=closeSQLObjects | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl[470] : MYSQL Closing SQL connection objects.
time=2022-02-24T11:41:00.794Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=getConnection | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl$SQLDriver[903] : MYSQL Recovered destination connection from cache (smartcity)
time=2022-02-24T11:41:00.796Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=getConnection | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl$SQLDriver[924] : MYSQL Pool status (smartcity) Max.: 3; Active: 1; Idle: 1
time=2022-02-24T11:41:00.797Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=createTable | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl[214] : MYSQL Executing SQL query 'create table if not exists `thing_dispD_thing`(recvTimeTs text,recvTime text,fiwareServicePath text,entityId text,entityType text,attrName text,attrType text,attrValue text,attrMd text)'
time=2022-02-24T11:41:00.817Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=closeSQLObjects | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl[470] : MYSQL Closing SQL connection objects.
time=2022-02-24T11:41:00.817Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=createTable | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl[226] : MYSQL Trying to add 'thing_dispD_thing' to the cache after table creation
time=2022-02-24T11:41:00.817Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=addTable | msg=com.telefonica.iot.cygnus.backends.sql.SQLCache[76] : 'thing_dispD_thing' added to the cache
time=2022-02-24T11:41:00.819Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=getConnection | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl$SQLDriver[903] : MYSQL Recovered destination connection from cache (smartcity)
time=2022-02-24T11:41:00.822Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=getConnection | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl$SQLDriver[924] : MYSQL Pool status (smartcity) Max.: 3; Active: 1; Idle: 1
time=2022-02-24T11:41:00.835Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=getFieldsForInsert | msg=com.telefonica.iot.cygnus.backends.sql.SQLQueryUtils[373] : [SQLQueryUtils.getFieldsForInsert] Preparing fields for insert for statement: (`recvTimeTs`,`recvTime`,`fiwareServicePath`,`entityId`,`entityType`,`attrName`,`attrType`,`attrValue`,`attrMd`)
time=2022-02-24T11:41:00.835Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=sqlInsertQuery | msg=com.telefonica.iot.cygnus.backends.sql.SQLQueryUtils[324] : [SQLQueryUtils.sqlInsertQuery] Preparing Insert query: INSERT INTO `thing_dispD_thing` (`recvTimeTs`,`recvTime`,`fiwareServicePath`,`entityId`,`entityType`,`attrName`,`attrType`,`attrValue`,`attrMd`) VALUES ('1645702859701','2022-02-24 11:40:59.701','/','thing:dispD','thing','TimeInstant','DateTime','2022-02-24T11:40:58.919Z','[]') 
time=2022-02-24T11:41:00.842Z | lvl=INFO | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=insertTransaction | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl[696] : MYSQL Finished transactions into database: smartcity 
 insertQuery: INSERT INTO `thing_dispD_thing` (`recvTimeTs`,`recvTime`,`fiwareServicePath`,`entityId`,`entityType`,`attrName`,`attrType`,`attrValue`,`attrMd`) VALUES ('1645702859701','2022-02-24 11:40:59.701','/','thing:dispD','thing','TimeInstant','DateTime','2022-02-24T11:40:58.919Z','[]') 
time=2022-02-24T11:41:00.845Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=insertTransaction | msg=com.telefonica.iot.cygnus.backends.sql.SQLBackendImpl[732] : MYSQL Trying to add 'smartcity' and 'thing_dispD_thing' to the cache after insertion
time=2022-02-24T11:41:00.845Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=addDataBase | msg=com.telefonica.iot.cygnus.backends.sql.SQLCache[52] : 'smartcity' not added to the cache, since already existing
time=2022-02-24T11:41:00.845Z | lvl=DEBUG | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=addTable | msg=com.telefonica.iot.cygnus.backends.sql.SQLCache[72] : 'thing_dispD_thing' not added to the cache, since already existing
time=2022-02-24T11:41:00.846Z | lvl=INFO | corr=294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1 | trans=5ab56587-4510-4342-9fd6-e689d231ee62 | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=processNewBatches | msg=com.telefonica.iot.cygnus.sinks.NGSISink[624] : Finishing internal transaction (294eaa0b-bcbc-4802-96e9-fa52517921a7; cbnotif=1, ) Sink: mysql-sink Destination: smartcity_/_thing:dispD_thing
time=2022-02-24T11:41:06.417Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=processNewBatches | msg=com.telefonica.iot.cygnus.sinks.NGSISink[531] : Batch accumulation time reached, the batch will be processed as it is
time=2022-02-24T11:41:11.418Z | lvl=DEBUG | corr= | trans= | srv= | subsrv= | comp= | op=run | msg=org.apache.flume.node.PollingPropertiesFileConfigurationProvider$FileWatcherRunnable[131] : Checking file:/opt/apache-flume/conf/agent.conf for changes
time=2022-02-24T11:41:11.631Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=processNewBatches | msg=com.telefonica.iot.cygnus.sinks.NGSISink[531] : Batch accumulation time reached, the batch will be processed as it is
time=2022-02-24T11:41:11.679Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=processNewBatches | msg=com.telefonica.iot.cygnus.sinks.NGSISink[531] : Batch accumulation time reached, the batch will be processed as it is
time=2022-02-24T11:41:12.378Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[215] : [nmi] The configuration has not changed
time=2022-02-24T11:41:12.412Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[215] : [nmi] The configuration has not changed
time=2022-02-24T11:41:12.412Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[215] : [nmi] The configuration has not changed
time=2022-02-24T11:41:12.439Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=run | msg=com.telefonica.iot.cygnus.interceptors.NGSINameMappingsInterceptor$PeriodicalNameMappingsReader[215] : [nmi] The configuration has not changed
time=2022-02-24T11:41:15.293Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=processNewBatches | msg=com.telefonica.iot.cygnus.sinks.NGSISink[531] : Batch accumulation time reached, the batch will be processed as it is
time=2022-02-24T11:41:15.417Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=processNewBatches | msg=com.telefonica.iot.cygnus.sinks.NGSISink[531] : Batch accumulation time reached, the batch will be processed as it is
time=2022-02-24T11:41:15.847Z | lvl=DEBUG | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=processNewBatches | msg=com.telefonica.iot.cygnus.sinks.NGSISink[531] : Batch accumulation time reached, the batch will be processed as it is

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/LogHandlers.java [hunk 1]
```diff
@@ -1043,7 +1095,7 @@ public static void putLogLevel(HttpServletRequest request, HttpServletResponse r
         
         try {
             CommonConstants.LoggingLevels.valueOf(logLevel.toUpperCase());
-            LogManager.getRootLogger().setLevel(Level.toLevel(logLevel.toUpperCase()));
+            setLevel(LogManager.getRootLogger(), Level.toLevel(logLevel.toUpperCase()));
```
Comment (AlvaroVega) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/LogHandlers.java:1098]: This should be updated having in account: https://github.com/telefonicaid/fiware-cygnus/pull/2170

**Comment by fgalan** [comment]
Overpassed by PR https://github.com/telefonicaid/fiware-cygnus/pull/2212. Closing.

**Pull request closed without merging**